app/src/main/java/com/ataulm/whatsnext/WhatsNextService.java

```diff
@@ -1,75 +1,84 @@
 package com.ataulm.whatsnext;
 
-import com.ataulm.support.Clock;
-import com.ataulm.whatsnext.api.AuthenticationError;
-import com.ataulm.whatsnext.api.AuthenticationError.Type;
-import com.ataulm.whatsnext.api.Letterboxd;
+import com.ataulm.whatsnext.api.ApiFilm;
+import com.ataulm.whatsnext.api.ApiFilmRelationship;
+import com.ataulm.whatsnext.api.ApiSearchResponse;
+import com.ataulm.whatsnext.api.AuthTokenApiResponse;
+import com.ataulm.whatsnext.api.FilmRelationshipConverter;
+import com.ataulm.whatsnext.api.FilmSummaryConverter;
+import com.ataulm.whatsnext.api.LetterboxdApi;
 
-import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.Callable;
 
 import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Function;
 
 public class WhatsNextService {
 
-    private final Letterboxd letterboxd;
-    private final TokensStore tokensStore;
-    private final Clock clock;
+    private final LetterboxdApi letterboxdApi;
+    private final FilmSummaryConverter filmSummaryConverter;
+    private final FilmRelationshipConverter filmRelationshipConverter;
 
-    public WhatsNextService(Letterboxd letterboxd, TokensStore tokensStore, Clock clock) {
-        this.letterboxd = letterboxd;
-        this.tokensStore = tokensStore;
-        this.clock = clock;
+    public WhatsNextService(LetterboxdApi letterboxdApi, FilmSummaryConverter filmSummaryConverter, FilmRelationshipConverter filmRelationshipConverter) {
+        this.letterboxdApi = letterboxdApi;
+        this.filmSummaryConverter = filmSummaryConverter;
+        this.filmRelationshipConverter = filmRelationshipConverter;
     }
 
     public Observable<Token> login(final String username, final String password) {
```

Comment:
I guess these calls on the "service" were all `Observable` rather than single because I expected to use some sort of local cache. Although retrofit can handle `Observable` (I think), it's more correct to use `Single` on the API declaration, and convert this to observable here.

Whether this _should_ be observable wasn't important in this PR, and I wanted to touch as little as possible.